### PR TITLE
Added Dockerfile for local development

### PR DESCRIPTION
**Pull request by plusjeff**
This should make it super easy for folks to get setup to do development locally. 

@paultag : This is the first Dockerfile I've ever made, so definitely would love your review, as I'm not savvy with Docker best practices, etc. Also, this probably doesn't meet the need to have a Dockerfile act simultaneously as a dev and deployment image. That'd be cool to do in a future PR (and would love to see how you or someone else would tackle that).

@orischwartz : This should make it dramatically easier to get setup on Windows for local development. Can you give it a whirl to see if the instructions work smoothly on Windows?

See the **Quick Developer Setup (w/ Docker)** section of the updated [README.markdown](https://github.com/department-of-veterans-affairs/caseflow/blob/docker/README.markdown) file for instructions.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Install Dependencies
+RUN yum -y update
+RUN yum -y install git                  # For getting Ruby gems and Rbenv
+RUN yum -y install java-1.8.0-openjdk   # For running JRuby
+RUN yum -y install tar                  # For unpacking JRuby
+RUN yum -y install "gcc-c++"            # For building Ruby dependencies
+RUN yum -y install wget                 # For getting PDFTK
+RUN yum -y install libgcj               # For PDFTK
+RUN yum -y install which                # For detecting PDFTK
```
Comment (paultag) [inline Dockerfile]: I'd do all the installs on a single line; this creates a bunch of new layers

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ENV PATH="$RBENV_HOME/bin:$RBENV_HOME/shims:$PATH"
+RUN git clone https://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build
+
+# Install JRuby and bundler
+RUN rbenv install jruby-9.0.3.0
+RUN rbenv global jruby-9.0.3.0
+RUN gem install bundler
+
+# Pre-install Caseflow dependencies
+RUN git clone https://github.com/department-of-veterans-affairs/caseflow.git /caseflow
```
Comment (paultag) [inline Dockerfile]: perhaps do an `ADD . /caseflow` at the top

Comment (paultag) [inline Dockerfile]: Erm, nevermind, here's fine (but `ADD`)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+WORKDIR "/caseflow"
+RUN bundle install
+RUN rm -rf /caseflow/*
```
Comment (paultag) [inline Dockerfile]: What's this for?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+# Configure Test Mode
+ENV CASEFLOW_TEST=1
+EXPOSE 3000
+
+# Start Caseflow (from the mounted directory)
+## Sample start command:
+## docker run -it -v "$(pwd)":/caseflow -p 3000:3000 caseflow
+CMD bundle install && rails s --binding 0.0.0.0 --port 3000
```
Comment (paultag) [inline Dockerfile]: why bundle install before running rails?

**Comment by plusjeff** [comment]
Sorry! Never found time to come back to this one, but think it's probably not relevant anymore. Will close w/o merge.

**Pull request closed without merging**